packages/compass-settings/src/stores/settings.ts

```diff
@@ -52,9 +63,43 @@ type SaveSettingsAction = {
   type: ActionTypes.SettingsSaved;
 };
 
-export type Actions = SettingsFetchedAction | SaveSettingsAction;
+type SettingsFetchStartAction = {
+  type: ActionTypes.SettingsFetchedStart;
+};
+
+type OpenSettingsModalAction = {
+  type: ActionTypes.OpenSettingsModal;
+  tab?: SettingsTabId;
+};
+
+type SelectTabAction = {
+  type: ActionTypes.SelectTab;
+  tab?: SettingsTabId;
+};
+
+type CloseSettingsModalAction = {
+  type: ActionTypes.CloseSettingsModal;
+};
+
+type ChangeFieldValueAction<K extends keyof UserConfigurablePreferences> = {
+  type: ActionTypes.ChangeFieldValue;
+  field: K;
+  value: UserConfigurablePreferences[K];
+};
+
+export type Actions =
+  | SettingsFetchedAction
+  | SaveSettingsAction
+  | SettingsFetchStartAction
+  | OpenSettingsModalAction
+  | SelectTabAction
+  | CloseSettingsModalAction
+  | ChangeFieldValueAction<keyof UserConfigurablePreferences>;
 
-export const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
+export const reducer: Reducer<State, Actions> = (
```

Comment:
I know it's a pain in the butt, but Redux docs [recommend against unions](https://redux.js.org/usage/usage-with-typescript#avoid-action-type-unions) for action type and we generally follow this rule by using [isAction](https://github.com/mongodb-js/compass/blob/9b0315879799f8d884944b6779b44976609f2e2a/packages/atlas-service/src/store/atlas-signin-reducer.ts#L8-L13) helper

Reply:
Ah, right. I literally forgot that that's what we do in Compass tbh 🙈 Just saw that `action` wasn't typed and wanted to fix that, but I'll gladly remove this again and use `isAction` instead, will do that in the next commit.

It does make me wonder, should we maybe prefer typing our reducers with `Reducer<State, Action>` (instead of the implied default `Reducer<State, AnyAction>`)? That would still make sure that we're not accidentally relying on the `any`-ness of the `action` parameter we're using

Reply:
Done in 8236817e9e55c318f7e578a80722b8310c0f7742 :+1: